schism/palettes.c

```diff
@@ -314,10 +314,30 @@ uint8_t current_palette[16][3] = {
 	/* 14 */ {63, 63, 21},
 	/* 15 */ {63, 63, 63},
 };
+
+uint8_t user_palette[16][3] = {
```

Comment:
You can just put this in the front of `palettes` and use it through
```c
#define USER_PALETTE (palettes[0])
```
which should make the current palette functions work just fine.

Reply:
That doesn't look right. We don't want the user palette to override the "Light Blue" palette. And also `palettes[0]` is not the same type.

Reply:
I think I explained what I was thinking in a confusing way lol, my bad

I was saying that it would be nicer to put the user-defined palette into the regular palettes list with the name "User Defined", then use a helper macro `USER_PALETTE` to access it in case it ever moves or whatnot. That way the functions can just pass over the array `palettes` without having to worry about the existence of the user palette at all.

Reply:
Ohhh yeah that's a good idea then we don't need so many checks for user palette 😆